hudi-spark/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -111,6 +111,9 @@ private[hudi] object HoodieSparkSqlWriter {
         tableConfig = tableMetaClient.getTableConfig
       }
 
+      val metaClient = new HoodieTableMetaClient(sparkContext.hadoopConfiguration, path.get)
+      val commitActionType = DataSourceUtils.getCommitActionType(operation, metaClient)
```

Comment:
should we have a better way of getting the commit action type? I am bit concerned about creating new metaClient just for this.

Reply:
We just need tableType to create commit/deltacommit in the default case. I moved this to a static method in util class and removed dependency on MetaClient.